src/common/Microsoft.DotNet.UpgradeAssistant.Abstractions/IProject.cs

```diff
@@ -21,6 +21,8 @@ public interface IProject
 
         IEnumerable<IProject> ProjectReferences { get; }
 
+        public IEnumerable<string> AllProjectReferences { get; }
```

Comment:
Is there a way we can accommodate the goal of your PR w/o breaking this public interface?
I don't mind if this gets added to `MSBuildProject` but must be _very_ careful about this change as we ship `.Abstractions` as one of our only public nuget pkgs for extending UA. So I'd love to see if there's a way you can do your work w/o modifying this interface

Reply:
Since `MSBuildProject` is not available in the windows extension. `IProject` and `IProjectFile` are 2 places where this can reside. Ideally, IProjectFile should have exposed all project properties and items through some iterative property but it does not. So. at the moment there is no good way to access some items like vcxproj project references through any of these interfaces.
I am ok to discuss this and implement this in a better way if you have a suggestion.

Reply:
One way to do this could be making this change through the try-convert module

Reply:
Since you need this for Windows project only... could we do `WindowsProject : IProject` and add the property there?

Reply:
Have a look [at this commit](https://github.com/dotnet/upgrade-assistant/commit/555cea49ebd00be7e18fddb9b06986b7d23c8f6b)